src/RulesDeploy.js

```diff
@@ -51,21 +51,50 @@ RulesDeploy.prototype = {
   /**
    * Create rulesets for each file added to this deploy, and record
    * the name for use in the release process later.
+   *
+   * If the ruleset to create is identical to the latest existing ruleset,
+   * then we record the existing ruleset name instead of creating a duplicate.
+   *
+   * @param {string} service The service to create a ruleset.
+   *    Valid values are "cloud.firestore" and "firebase.storage"
    */
-  createRulesets: function() {
+  createRulesets: function(service) {
     var self = this;
-    var promises = [];
-    _.forEach(this.rulesFiles, function(files, filename) {
-      utils.logBullet(
-        clc.bold.cyan(self.type + ":") + " uploading rules " + clc.bold(filename) + "..."
-      );
-      promises.push(
-        gcp.rules.createRuleset(self.options.project, files).then(function(rulesetName) {
-          self.rulesetNames[filename] = rulesetName;
-        })
-      );
-    });
-    return Promise.all(promises);
+    var latestRulesetName = "";
+    return gcp.rules
+      .getLatestRulesetName(self.options.project, service)
+      .then(function(retrievedLatestRulesetName) {
+        if (_.isNull(retrievedLatestRulesetName)) {
```

Comment:
1) having a special case for `null` is generally discouraged. Since `null` is falsey, simply doing `!retrievedLastRulsetName` is sufficient (though, do note, in JS, an empty string `""` is also falsey - that's really the only case you'd want to write more specific checks, and even then it's "special casing" empty string)
2) since you return in your `if` statement, including `else` below is redundant. Remove the `else` and just leave the else logic at the same level as the `if`

Reply:
Done